doc/specs/vulkan/chapters/fundamentals.txt

```diff
@@ -754,14 +766,14 @@ commands, which will silently ignore these values.
 [[fundamentals-validusage-pointers]]
 ==== Valid Usage for Pointers
 
-Any parameter that is a pointer must: either be a valid pointer, or if
-_explicitly called out in the specification_, can: be `NULL`.
+A Valid Usage section always explicitly states if a command parameter or a
+structure member is required to be a valid pointer.
+
 A pointer is valid if it points at memory containing values of the number
 and type(s) expected by the command, and all fundamental types accessed
 through the pointer (e.g. as elements of an array or as members of a
 structure) satisfy the alignment requirements of the host processor.
```

Comment:
I should probably say something about the `void*` here; perhaps:

> Note: Pointer to a `void` type is either a byte array (if it has explicit expected size), or plain integer (i.e. unlike other pointers it has default expected size of 0 and expected type "nothing").

Reply:
I'm not sure what that would add - especially given that vkUpdateDescriptorSetWithTemplateKHR already violates this - it's not "just" a byte array. Also saying "it's an int" doesn't really make sense in the way I think you mean it to. I don't think we should add such a note in any capacity. The byte array stuff is kind of implicit by the nature of it being specified as a C99 API as well.

Reply:
Right. TBH I am half-tempted to delete most of this chapter, because it is mostly now explicit anyway since the autogenerator was made.

Reply:
Yea I mean this section was added as a sort of "guideline" to how to read the VU statements, rather than as a normative "spec works this way" section - because we knew full well it didn't (and still doesn't) entirely adhere to this.

Reply:
Yeah, no. Disregard.
It adds nothing. It is fully covered by VU sections as defined below.
"Optional void*" has no VU (or lacks word "valid"),
array of bytes are explicitly in VUs too,
and `void*` without optional, len or noautovalidity should not happen (it would be either `char` instead or complete nonsense).